server/src/main/java/org/candlepin/model/ProductAttribute.java

```diff
@@ -163,6 +163,17 @@ public int hashCode() {
     }
 
     /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
+        return this.hashCode();
```

Comment:
@Ceiu The java implementation hash no guarantee of consistency. See https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode()

> Whenever it is invoked on the same object more than once during an execution of a Java application, the hashCode method must consistently return the same integer, provided no information used in equals comparisons on the object is modified. This integer need not remain consistent from one execution of an application to another execution of the same application.